src/random-seed/random-seed.c

```diff
@@ -277,6 +290,17 @@ static int run(int argc, char *argv[]) {
                                                        "Got EOF while reading from /dev/urandom.");
                 }
 
+                /* If we previously read in a seed file, then hash the new seed into the old one,
+                 * and replace the last 32 bytes of the seed with the hash output, so that the
+                 * new seed file can't regress in entropy. */
+                if (read_seed_file) {
+                        sha256_process_bytes(&k, sizeof(k), &hash_state);
+                        sha256_process_bytes(buf, k, &hash_state);
+                        sha256_finish_ctx(&hash_state, hash);
+                        l = MIN(k, 32);
+                        memcpy((uint8_t *)buf + k - l, hash, l);
```

Comment:
doesn't matter, but if you'd hash this into the beginning of the buffer, you don't need a temporary variable, i.e.:

```c
memcpy(buf, hash, MIN(32, k));
```

And whether the SHA256 result is placed at the end or the beginning of the seed buffer doesn't really matter, does it?

Reply:
It doesn't really matter at all here I suppose. Were this a different sort of thing, maybe we'd have some handwavy concerns about hiding the last bytes that were hashed rather than the first, to make length extension or other attacks harder. But that's very much not what's happening here, so I guess we could do the beginning if you feel strongly about it.